apps/vc-api/test/vc-api/exchanges/consent-and-resident-card-credentialexchange/consent-and-resident-card.e2e-suite.ts

```diff
@@ -0,0 +1,320 @@
+    const issuanceVpRequest = await walletClient.startExchange(issuanceExchangeEndpoint, exchange.queryType);
+    const issuanceExchangeContinuationEndpoint = getContinuationEndpoint(issuanceVpRequest);
+    expect(issuanceExchangeContinuationEndpoint).toContain(issuanceExchangeEndpoint);
+
+    // As holder, create new DID and presentation to authentication as this DID
+    // DID auth presentation: https://github.com/spruceid/didkit/blob/c5c422f2469c2c5cc2f6e6d8746e95b552fce3ed/lib/web/src/lib.rs#L382
+    const holderDIDDoc = await walletClient.createDID('key');
+    const holderVerificationMethod = holderDIDDoc.verificationMethod[0].id;
+    const options: ProvePresentationOptionsDto = {
+      verificationMethod: holderVerificationMethod,
+      proofPurpose: ProofPurpose.authentication,
+      challenge: issuanceVpRequest.challenge
+    };
+    const didAuthResponse = await request(app.getHttpServer())
+      .post(`${vcApiBaseUrl}/presentations/prove/authentication`)
+      .send({ did: holderDIDDoc.id, options })
+      .expect(201);
+    const didAuthVp = didAuthResponse.body;
+    expect(didAuthVp).toBeDefined();
+
+    // As holder, continue exchange by submitting did auth presention
+    for (let i = 0; i < numHolderQueriesPriorToIssuance; i++) {
+      await walletClient.continueExchange(issuanceExchangeContinuationEndpoint, didAuthVp, true, true);
+    }
+    issuanceCallbackScope.done();
+
+    // As the issuer, get the transaction
+    // TODO TODO TODO!!! How does the issuer know the transactionId? -> Maybe can rely on notification
+    const urlComponents = issuanceExchangeContinuationEndpoint.split('/');
+    const transactionId = urlComponents.pop();
+    const transaction = await walletClient.getExchangeTransaction(exchange.getExchangeId(), transactionId);
+
+    // As the issuer, check the result of the transaction verification
+    expect(transaction.presentationSubmission.verificationResult.checks).toContain('proof');
+    expect(transaction.presentationSubmission.verificationResult.errors).toHaveLength(0);
+
+    // As the issuer, create a presentation to provide the credential to the holder
+    const holderKeyId = holderDIDDoc.verificationMethod[0].publicKeyJwk.kid;
+    const issueResultConsentCredential = await exchange.issueConsentCredential(holderKeyId, walletClient);
+    const issuedVPConsentCredential = issueResultConsentCredential.vp;
+    const issueResultResidentiCard = await exchange.issueResidentCardCredential(didAuthVp, walletClient);
+    const issuedVPResidentCard = issueResultResidentiCard.vp; // VP used to wrapped issued credentials
+
+    const finalPresentation: Presentation = {
+      '@context': ['https://www.w3.org/2018/credentials/v1'],
+      type: ['VerifiablePresentation'],
+      verifiableCredential: [
+        issuedVPConsentCredential.verifiableCredential[0],
+        issuedVPResidentCard.verifiableCredential[0]
+      ]
+    };
```

Comment:
I think this is the presentation that is returned to the holder from the issuer, right? If so, the presentation doesn't need to contain the consent credential as we can imagine the holder can hold it in their wallet after self-signing (after the `exchange.issueConsentCredential`). So the holder presentation can just reference this credential without needing to get it from an issuer, IMO.
```suggestion
    const finalPresentation: Presentation = {
      '@context': ['https://www.w3.org/2018/credentials/v1'],
      type: ['VerifiablePresentation'],
      verifiableCredential: [
        issuedVPResidentCard.verifiableCredential[0]
      ]
    };
```